third_party/helm/helm.go

```diff
@@ -0,0 +1,208 @@
+)
+
+type Opts struct {
+	// Name is used to indicate the name of the helm chart being processed
+	Name string
+	// Namespace is used to indicate the namespace in which the helm chart
+	// identified by Name will be processed
+	Namespace string
+	// ReleaseName is used to indicate the name of the release to be used
+	// while installing or upgrading the chart identified by Name into the
+	// Namespace
+	ReleaseName string
+	// Version is the helm chart version that should be deployed to the test
+	// infrastructure
+	Version string
+	// Chart is used to indicate the full path of the .tgz artifact of the Helm
+	// chart in case if your tests require you to be able to deploy a packaged
+	// helm chart that is available locally as the tarball
+	Chart string
+	// Mode is used to indicate the mode in which the helm operation is being
+	// performed. These are the first level commands that are processed by the
+	// helm Binary
+	Mode string
+	// Args is used to pass any additional arguments that you might want to pass
+	// for running the helm command in question
+	Args []string
+	// Wait is used to indicate if the helm command should wait for the runtime
+	// to reach an acceptable state before returning the control back. The duration
+	// for which this call is blocked is defined by the value set to Timeout
+	Wait bool
+	// Timeout is used to indicate the time to wait for any individual Kubernetes ops
+	Timeout string
+}
+
+type Manager struct {
+	e          *gexe.Echo
+	kubeConfig string
+}
+
+type Option func(*Opts)
+
+const (
+	missingHelm = "'helm' command is missing. Please ensure the tool exists before using the helm manager"
+)
+
+// WithName is used to set the name of the helm chart being processed
+func WithName(name string) Option {
+	return func(opts *Opts) {
+		opts.Name = name
+	}
+}
+
+// WithNamespace is used to configure the namespace in which the helm chart
+// identified by WithName will be processed
+func WithNamespace(namespace string) Option {
+	return func(opts *Opts) {
+		opts.Namespace = namespace
+	}
+}
+
+// WithReleaseName is used to configure the name of the release to be used
+// while installing or upgrading the chart identified by WithName into the
+// WithNamespace configured namespace
+func WithReleaseName(releaseName string) Option {
+	return func(opts *Opts) {
+		opts.ReleaseName = releaseName
+	}
+}
+
+// WithVersion is used to configre helm chart version that should be deployed
+// to the test infrastructure
+func WithVersion(version string) Option {
+	return func(opts *Opts) {
+		opts.Version = version
+	}
+}
+
+// WithChart is used to configure the full path of the .tgz artifact of the Helm
+// chart in case if your tests require you to be able to deploy a packaged
+// helm chart that is available locally as the tarball
+func WithChart(chart string) Option {
+	return func(opts *Opts) {
+		opts.Chart = chart
+	}
+}
+
+// WithMode is used to invoke the first sub command of the helm binary. This can
```

Comment:
As mentioned earlier, I don't think mode should be open-ended.  The API should have specific modes supported when creating tests.  I would suggest that the supported modes should be exposed as a method in the API as follows

```go
helm.New(configFile).RunRepo(WithName(...), WithArgs(...), etc)
helm.New(configFile).RunInstall(WithName(...), WithArgs(...), etc)
helm.New(configFile).RunTest(WithName(...), WithArgs(...), etc)
```